libs/model/src/community/CreateStakeTransaction.command.ts

```diff
@@ -0,0 +1,128 @@
+import type { Command } from '@hicommonwealth/core';
+import { commonProtocol, schemas } from '@hicommonwealth/core';
+import Web3 from 'web3';
+import { models } from '../database';
+import { mustExist } from '../middleware/guards';
+
+/**
+ * This function will first search the database for existing transactions,
+ * and create and persist all remaining transactions
+ * @constructor
+ */
+export const CreateStakeTransaction: Command<
+  typeof schemas.commands.CreateStakeTransaction
+> = () => ({
+  ...schemas.commands.CreateStakeTransaction,
+  auth: [],
+  body: async ({ payload }) => {
+    const { transaction_hash } = payload;
+
+    // Find transactions that already exist in database
+    const existingTransactions = await models.StakeTransaction.findOne({
+      where: {
+        transaction_hash: transaction_hash,
+      },
+    });
+
+    if (existingTransactions) {
+      return existingTransactions.toJSON();
+    }
+
+    // newTransactionIds are the remaining transactions that we must query web3 for.
+    const community = await models.Community.findOne({
+      where: { id: payload.community_id },
+      include: [
+        {
+          model: models.ChainNode,
+          attributes: ['eth_chain_id', 'url'],
+        },
+      ],
+    });
+
+    mustExist('Community', community);
+    mustExist('Chain Node', community!.ChainNode);
+    mustExist('Community namespace', community!.namespace);
+
+    if (
```

Comment:
we have newNamespaceValidator in protocol services... let's try to consolidate all protocol validations in a single place... can be extended as needed